build-tools/packages/build-infrastructure/src/test/fluidRepo.test.ts

```diff
@@ -104,3 +108,52 @@ describe("loadFluidRepo", () => {
 		});
 	});
 });
+
+describe("setDependencyRange", () => {
+	const repo = loadFluidRepo(testRepoRoot);
+	const main = repo.releaseGroups.get("main" as ReleaseGroupName);
+	assert(main !== undefined);
+
+	const group2 = repo.releaseGroups.get("group2" as ReleaseGroupName);
+	assert(group2 !== undefined);
+
+	const secondWorkspace = repo.workspaces.get("second" as WorkspaceName);
+	assert(secondWorkspace !== undefined);
+
+	afterEach(async () => {
+		await git.checkout(["HEAD", "--", testRepoRoot]);
+		repo.reload();
+	});
+
+	it("updates the dependency range given release group package dependencies", async () => {
+		const group2Version = semver.parse(group2.version);
+		assert(group2Version !== null);
+		await setDependencyRange(main.packages, group2.packages, group2Version);
+		const allCorrect = main.packages
+			.filter((pkg) => group2.packages.some((group2Pkg) => group2Pkg.name === pkg.name))
+			.every((pkg) => pkg.version === group2.version);
+		expect(allCorrect).to.be.true;
+	});
+
+	it("updates the dependency range given workspace package dependencies", async () => {
+		const workspace2Version = semver.parse(secondWorkspace.rootPackage.version);
+		assert(workspace2Version !== null);
+		await setDependencyRange(main.packages, secondWorkspace.packages, workspace2Version);
+		const allCorrect = main.packages
+			.filter((pkg) =>
+				group2.packages.some((workspace2Pkg) => workspace2Pkg.name === pkg.name),
+			)
+			.every((pkg) => pkg.version === secondWorkspace.rootPackage.version);
+		expect(allCorrect).to.be.true;
+	});
+
+	it("updates the dependency range to explicit version", async () => {
+		const version = semver.parse("2.0.0");
+		assert(version !== null);
```

Comment:
No need to change this, I just want to clarify some earlier feedback. It's a little more OK to use `semver.parse("2.0.0")!` in this case because the value being parsed is a constant. When you're taking in some data that _should_ be a valid semver, it's best to check that it actually is. But when you're parsing a hardcoded value, it's more OK (in my opinion) to use a `!` in test code at least.